datacube_stats/main.py

```diff
@@ -109,19 +111,33 @@ def gather_tile_indexes(tile_index, tile_index_file):
               help="A file consisting of tile indexes specified as [X] [Y] per line")
 @click.option('--output-location', help='Override output location in configuration file')
 @click.option('--year', type=int, help='Override time period in configuration file')
+@click.option('--task-slice', type=str, default='::',
```

Comment:
It is possible to tell `click` to handle custom types, so rather than it being an unparsed string that you then parse (and it might fail to parse) it's best to instead parse it before `main` that way you can communicate parsing errors to the user in a more convenient time.

See for example this: https://github.com/GeoscienceAustralia/digitalearthau/blob/c3c048dfdccf1b7205a9af5378846f5cdd6fbcce/digitalearthau/qsub.py#L187-L209

It does require some boiler plate, but is more useable.